gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/data/ImageDownloader.kt

```diff
@@ -0,0 +1,71 @@
+package com.gravatar.quickeditor.data
+
+import android.app.DownloadManager
+import android.content.Context
+import android.content.pm.PackageManager
+import android.net.Uri
+import android.os.Environment
+import com.gravatar.BuildConfig
+import com.gravatar.services.GravatarResult
+import java.net.URI
+
+internal class ImageDownloader(
+    private val context: Context,
+) {
+    private val downloadManager: DownloadManager? = context.getSystemService(DownloadManager::class.java)
+    private val appName: String by lazy {
+        val applicationInfo = context.packageManager.getApplicationInfo(
+            context.packageName,
+            PackageManager.GET_META_DATA,
+        )
+        context.packageManager.getApplicationLabel(applicationInfo).toString()
+    }
+
+    fun downloadImage(imageUrl: URI): GravatarResult<Unit, DownloadManagerError> {
+        return when {
+            downloadManager == null -> GravatarResult.Failure(DownloadManagerError.DOWNLOAD_MANAGER_NOT_AVAILABLE)
+            !isDownloadManagerEnabled() -> GravatarResult.Failure(DownloadManagerError.DOWNLOAD_MANAGER_DISABLED)
+            else -> {
+                val request = DownloadManager.Request(Uri.parse(imageUrl.withMaxSizeQueryParam().toString()))
+                    .addRequestHeader("X-Platform", "Android")
+                    .addRequestHeader("X-SDK-Version", BuildConfig.SDK_VERSION)
+                    .addRequestHeader("X-Source", appName)
+                    .setNotificationVisibility(DownloadManager.Request.VISIBILITY_VISIBLE_NOTIFY_COMPLETED)
+                    .setMimeType("image/*")
+                    .setDestinationInExternalPublicDir(
+                        Environment.DIRECTORY_DOWNLOADS,
+                        "gravatar_image_${System.currentTimeMillis()}.png",
```

Comment:
⛏ Totally nitpicking. Maybe we can try to use the same name for the same image so in the download folder, we'll see that it's the same image (the system adds the file-name-2.png

As I said, we can consider this in a different PR if we think it looks better.

Reply:
I thought about it and tried it, but it didn't work like that. 
The image had the same file name... Instead of sth like `filename(1).png` it was `filename.png(1)`. I can try on different devices, but still, the end result was the same, multiple files were created.

Reply:
Maybe different devices handle this in other ways. Nevermind then. 👍